### PR TITLE
First steps to delete flask_sqlalchemy

### PR DESCRIPTION
**Pull request by sloschert**
**First steps to delete flask_sqlalchemy**

This commit contains the first  steps to delete the dependency on flask_sqlalchemy.  Most importantly, it uses SQLAlchemy directly in our model definitions in `arbeitszeit_flask.database.models.py`. Consequently, some changes were necessary in `arbeitszeit_flask.database.repositories.py`. The syntax of how we query database tables changes as well as how all tables get created and dropped in our integration tests or in benchmarking.

**Replace Flask-Migrate with alembic**

Before this commit our database migrations would depend on flask_migrate which depends on flask_sqlalchemy. We delete this dependency and are using plain alembic now.

We still have a slight dependency between alembic and Flask: In `arbeitszeit_flask.migrations.env.py` we make use of a Flask app object to retrieve the correct db uri. This is a dependency we can get rid of in the near future.

The github job has been adjusted to use alembic instead of Flask-Migrate.

### CODE REVIEW

**Comment by sloschert** [comment]
Our database migrations are still dependent on flask_sqlalchemy, that's why one check is (correctly) failing. I'll fix it.

**Review by Amittai-Aviram — COMMENTED** [review]
This is a great change overall.  I do have one question — about connections.  I do not know enough about this technology to understand how the connection to the DB server is managed.  In particular, is there a single, persistent connection that is shared by all DB-related code over a session (defined by a user logged in), or do we create a new connection every time we need to interact with the DB engine?  When I looked for "connect" in your changes, I saw code that looked more like the latter — e.g., a new connection is created just for a migration — but that is just my guess, and you wouuld know better than I.  Also, does your change set cause a change in this underlying behavior?  Thanks!

File: alembic.ini [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# A generic, single database configuration.
```
[the remark this answers is not in the document]
Comment (sloschert) [reply alembic.ini:1]: No, I created these settings (also the `env.py` file) by calling `alembic init` and adapted them where necessary.
Nevertheless, the current settings are similar to the ones that we had before (which were generated by Flask-Migrate).

**Comment by sloschert** [comment]
> This is a great change overall. I do have one question — about connections. I do not know enough about this technology to understand how the connection to the DB server is managed. In particular, is there a single, persistent connection that is shared by all DB-related code over a session (defined by a user logged in), or do we create a new connection every time we need to interact with the DB engine? When I looked for "connect" in your changes, I saw code that looked more like the latter — e.g., a new connection is created just for a migration — but that is just my guess, and you wouuld know better than I. Also, does your change set cause a change in this underlying behavior? Thanks!

Probably you are refering to the code in `arbeitszeit_flask/migrations/env.py`:

```    
connectable = create_engine(get_db_uri())

with connectable.connect() as connection:
    ....
```
This code is only run when we run migrations. It uses the `create_engine` function of sqlalchemy to create an engine.  The docstring of this function says:

"Once established, the newly resulting :class:`_engine.Engine` will
    request a connection from the underlying :class:`_pool.Pool` once
    :meth:`_engine.Engine.connect` is called, or a method which depends on it
    such as :meth:`_engine.Engine.execute` is invoked.   The
    :class:`_pool.Pool` in turn
    will establish the first actual DBAPI connection when this request
    is received.   The :func:`_sa.create_engine` call itself does **not**
    establish any actual DBAPI connections directly."

Note though, that the database engines and sessions that we use for our development server, the integration test server or our production server are (in the current branch) still managed by  Flask-SQLAlchemy, and not by SQLAlchemy directly! Consequently, in several places of our code we simply import a global database object: `from arbeitszeit_flask.extensions import db`.

I am not completely sure how Flask-SQLAlchemy is  handling connections and sessions, but it works probably like described on this section of the SQLAlchemy documentation on the `scoped_session` object:

https://docs.sqlalchemy.org/en/20/orm/contextual.html#using-thread-local-scope-with-web-applications

I have not noticed any change in the underlying behaviour, there should not be any.

**Review by Amittai-Aviram — APPROVED** [review]
LGTM!  Thank you for doing this and for your replies and explanations.

**Pull request merged**